### PR TITLE
Jesh + Patrick's finished LS-AJAX

### PR DESCRIPTION
**Pull request by jeshsy**
./Jesh and ./Patrick were "branch" folders
./friends has the final thing

... also Phoebe is best friend

### CODE REVIEW

**Comment by mixelpixel** [comment]
just commenting to see if get email notices

**Comment by mixelpixel** [comment]
...and agreed about Phoebe 😉

File: Jesh/friends/src/actions/index.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+};
+
+export const POST_FRIEND = 'POST_FRIEND';
+
+export const postFriend = (input) => {
+  const promise = axios.post('http://localhost:5000/new-friend', input);
+  return {
+    type: POST_FRIEND,
+    payload: promise
+  }
```
Comment (seanchen1991) [inline Jesh/friends/src/actions/index.js]: `;` here

File: Jesh/friends/src/components/FriendForm.js [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    this.handleChangeName = this.handleChangeName.bind(this);
+    this.handleChangeAge = this.handleChangeAge.bind(this);
+    this.handleChangeEmail = this.handleChangeEmail.bind(this);
+    this.addNewFriend = this.addNewFriend.bind(this);
+  }
+
+  handleChangeName(event) {
+    this.setState({
+      Name: event.target.value
+    })
```
Comment (seanchen1991) [inline Jesh/friends/src/components/FriendForm.js]: `;` here

File: Jesh/friends/src/components/FriendForm.js [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+  }
+
+  handleChangeAge(event) {
+    this.setState({
+      Age: event.target.value
+    })
```
Comment (seanchen1991) [inline Jesh/friends/src/components/FriendForm.js]: `;` here

File: Jesh/friends/src/components/FriendForm.js [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+  } 
+
+  handleChangeEmail(event) {
+    this.setState({
+      Email: event.target.value
+    })
```
Comment (seanchen1991) [inline Jesh/friends/src/components/FriendForm.js]: `;` here

File: Jesh/friends/src/components/FriendForm.js [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+          onChange={this.handleChangeAge}
+        />
+        Email:<input 
+          placeholder="Email"
+          value={this.state.Email}
+          onChange={this.handleChangeEmail}
+        />
+        <button type="submit" onClick={this.addNewFriend}>Submit Friend Request</button>
+      </form>
+    )
```
Comment (seanchen1991) [inline Jesh/friends/src/components/FriendForm.js]: `;` here

File: Jesh/friends/src/index.js [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+import App from './components/App';
+
+import friendsReducer from './reducers'; // <--- ./reducers/index.js
+
+
+// You will need to use `ReduxPromise` as a middleware inside of `src/index.js`.
+const createStoreWithMiddleware = applyMiddleware(ReduxPromise)(createStore);
+
+// const or let?????????
+const STORAGE_WAREHOUSE = createStoreWithMiddleware(friendsReducer, window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__());
```
Comment (seanchen1991) [inline Jesh/friends/src/index.js]: `const` here is fine, to answer the question in the comment.

**Comment by mixelpixel** [comment]
I blame all the missing semi-colons on Jesh. Er... Phoebe did it!

**Pull request closed without merging**